components/org.wso2.carbon.crypto.provider.hsm/src/main/java/org/wso2/carbon/crypto/provider/hsm/internal/HSMCryptoImplComponent.java

```diff
@@ -0,0 +1,170 @@
+import org.wso2.carbon.crypto.api.KeyResolver;
+import org.wso2.carbon.crypto.provider.hsm.HSMBasedExternalCryptoProvider;
+import org.wso2.carbon.crypto.provider.hsm.HSMBasedInternalCryptoProvider;
+import org.wso2.carbon.crypto.provider.hsm.HSMBasedKeyResolver;
+
+/**
+ * The class which is used to deal with the OSGi runtime for service registration and injection.
+ */
+@Component(
+        name = "org.wso2.carbon.crypto.provider.hsm",
+        immediate = true
+)
+public class HSMCryptoImplComponent {
+
+    private static final Log log = LogFactory.getLog(HSMCryptoImplComponent.class);
+    private static final String CRYPTO_SERVICE_ENABLING_PROPERTY_PATH = "CryptoService.Enabled";
+
+    private ServiceRegistration<ExternalCryptoProvider> hsmBasedExternalCryptoProviderServiceRegistration;
+    private ServiceRegistration<InternalCryptoProvider> hsmBasedInternalCryptoProviderServiceRegistration;
+    private ServiceRegistration<KeyResolver> hsmBasedKeyResolverServiceRegistration;
+    private ServerConfigurationService serverConfigurationService;
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        if (!isCryptoServiceEnabled()) {
+            if (log.isDebugEnabled()) {
+                log.debug("CryptoService is not enabled.");
+            }
+            return;
+        }
+
+        try {
+            BundleContext bundleContext = context.getBundleContext();
+            registerProviderImplementations(bundleContext);
+        } catch (Throwable e) {
+            String errorMessage = "An error occurred while activating HSM based crypto provider.";
+            if (log.isErrorEnabled()) {
+                log.error(errorMessage, e);
+            }
+        }
+
+        if (log.isInfoEnabled()) {
+            log.info("HSM Based crypto provider has been activated successfully.");
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context) {
+
+        hsmBasedExternalCryptoProviderServiceRegistration.unregister();
+        hsmBasedInternalCryptoProviderServiceRegistration.unregister();
+        hsmBasedKeyResolverServiceRegistration.unregister();
+    }
+
+    @Reference(
+            name = "serverConfigurationService",
+            service = ServerConfigurationService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            unbind = "unsetServerConfigurationService"
+    )
+    protected void setServerConfigurationService(ServerConfigurationService serverConfigurationService) {
+
+        this.serverConfigurationService = serverConfigurationService;
+    }
+
+    protected void unsetServerConfigurationService(ServerConfigurationService serverConfigurationService) {
+
+        this.serverConfigurationService = null;
+    }
+
+    protected boolean isCryptoServiceEnabled() {
+
+        String enabled = serverConfigurationService.getFirstProperty(CRYPTO_SERVICE_ENABLING_PROPERTY_PATH);
+        if (!StringUtils.isBlank(enabled)) {
+
```

Comment:
I think you don't need this complex logic within this method. Simply you can remove the whole method other than line 109 and then return Boolean.parseBoolean(enabled)